### ISSUE TITLE
0.9.4: build process clones gtest

### ISSUE THREAD

**Issue by kloczek** [iovisor/bpftrace#1129]
It would be good to switch to use system installed gtest because it will not work if build system is cut off from public network.

**Comment by danobi** [comment]
We do this b/c distros ship different/old versions of gtest and we rely on specific features. You only need network access the first time you build (to download). Subsequent builds should succeed offline.

**Comment by kloczek** [comment]
What is the problem to depend on at least some version using pkgconfig like other projects are doing that?

Again: this way of using gtest breaks build on the systems without connectivity to gtest git repo.

**Comment by caringi** [comment]
You can disable tests with the following cmake flag:

-DBUILD_TESTING:BOOL=OFF

**Comment by anatol** [comment]
Second on @kloczek request. The bpftrace should be consistent and use system libraries for its dependencies by default. If you concerned with version compatibility then add a check that requires gtest of some specific version (or newer).

**Comment by danobi** [comment]
@anatol so I better understand the tradeoff, why is it important that the tests use system gtest/gmock? Is it so packagers can run the tests during package time?

For context: vendored gtest/gmock is convenient for us b/c distros used to package it very strangely. For example, old versions of ubuntu only package gtest/gmock sources (which most of our CI runs on). Alpine still only packages shared gtest/gmock libs (so our static CI builds can't use system gtest/gmock either).

The bottom line is supporting system installed gtest/gmock introduces complexity into our already annoying complex build system. So I'd like to better understand the tradeoff I'm making here.

**Comment by kloczek** [comment]
> why is it important that the tests use system gtest/gmock?
In that cace there is no wasted time on compile and link one more time those libraries.

**Comment by danobi** [comment]
> In that cace there is no wasted time on compile and link one more time those libraries.

But are you building and running these tests during packaging?

Otherwise, as you know, vendored gtest/gmock is only built once. And you still have to link to system libraries just like you would to a vendored library. I can also assure you I've spent more "time" trying to do this than what anyone would save from that 1 time build cost of a vendored library.

**Comment by kloczek** [comment]
> But are you building and running these tests during packaging?
```
[tkloczko@barrel SPECS]$ grep ^%chec *| wc -l; ls -1 *spec | wc -l
2744
2831
```
Q Why not use included in source trees test suites to confirm that just build binaries (and other components installed in build env) are workikg corectly/as expected?
I think tjhhat only idiot concioselly would be trying to skip that part of the build processes.

**Comment by hailin84** [comment]
Is it possible to skip clone googletest code during making process? No network available.

**Comment by danobi** [comment]
@hailin84 see above: https://github.com/iovisor/bpftrace/issues/1129#issuecomment-582849064

ie

```
$ mkdir build
$ cd build
$ cmake .. -DBUILD_TESTING:BOOL=OFF
$ make
```

**Comment by anatol** [comment]
@danobi `-DBUILD_TESTING:BOOL=OFF` you provided disables tests. But the question was about avoiding git cloning the sources during tests.

**Comment by danobi** [comment]
> @danobi `-DBUILD_TESTING:BOOL=OFF` you provided disables tests. But the question was about avoiding git cloning the sources during tests.

Sure, currently there's no way to disable cloning gtest if you want to build the tests. Once https://github.com/iovisor/bpftrace/pull/1768 is merged, you'll be able to avoid gtest clone **and** build tests.

For now, if you want to avoid the gtest clone, you have to disable tests altogether.

**Comment by hailin84** [comment]
> @hailin84 see above: [#1129 (comment)](https://github.com/iovisor/bpftrace/issues/1129#issuecomment-582849064)
> 
> ie
> 
> ```
> $ mkdir build
> $ cd build
> $ cmake .. -DBUILD_TESTING:BOOL=OFF
> $ make
> ```

@danobi Thanks, It works.

**Comment by anatol** [comment]
Thank you @danobi!
